vulture_os/system/cluster/models.py

```diff
@@ -579,6 +579,36 @@ def save(self, *args, **kwargs):
         self.modified = timezone.now()
         return super().save(*args, **kwargs)
 
+    def await_result(self, interval=2, max_tries=30):
+        """
+
+        :param interval:    Time interval to wait before checking status again
+        :param max_tries:   The maximum number of times to check status
+        :return:
+                The result of the messagequeue or an empty string with an exception raised when the status
+                is not done in the interval and maximum tries given
+        """
+        counter = 0
+        try:
+            while True:
+                message_instance = MessageQueue.objects.get(pk=self.id)
+                if message_instance.status == "done":
+                    break
+                counter += 1
+                if counter == max_tries:
+                    raise APISyncResultTimeOutException(f"MessageQueue:: Timeout on the result of {message_instance.action}. Config is {message_instance.config}")
+                time.sleep(interval)
+
+            return message_instance.result
+        except APISyncResultTimeOutException as e:
+            logger.error(e)
+            return ""
+        except Exception as e:
+            logger.error(e)
+            return ""
```

Comment:
prefer letting the new exception bubble up on `await_result()` usages (don't forget to specify then what exceptions the function can raise)